### PR TITLE
Fix hazardous overloaded function in FwdSqlQuery

### PR DESCRIPTION
**Pull request by uklotzde**
When trying to store `double` values in the database all values != 0 ended up as 1.0. They were implicitly cast to `bool` because the wrong overloaded function was selected. The compiler didn't issue a warning about ambiguous casts.

I checked the code, this overload doesn't seem to be used yet. Nevertheless, targeted to master to avoid any unforeseeable new issues in the beta.

### CODE REVIEW

File: src/util/db/fwdsqlquery.h [hunk 1]
```diff
@@ -51,17 +51,21 @@ class FwdSqlQuery: protected QSqlQuery {
...
     void bindValue(const QString& placeholder, const QVariant& value) {
         QSqlQuery::bindValue(placeholder, value);
     }
 
-    // Overloaded function for type bool
-    void bindValue(const QString& placeholder, bool value) {
-        bindValue(placeholder, value ? QVariant(BOOLEAN_TRUE) : QVariant(BOOLEAN_FALSE));
+    // Specialized function for type bool
+    // This function can not be written safely as an overloaded
+    // version of bindValue() vor QVariant!! Otherwise numeric values
```
Comment (Holzhaus) [inline src/util/db/fwdsqlquery.h:64]:
```suggestion
    // version of bindValue() for QVariant!! Otherwise numeric values
```

**Review by Holzhaus — APPROVED** [review]
LGTM, thank you! Rebase on 2.3 or is master okay?

**Comment by uklotzde** [comment]
@Holzhaus Rebased and now safe for 2.3 and the future. Let's wait for CI.

**Comment by uklotzde** [comment]
Another force push. The `QVariant` version works as expected and the overload for `bool` was both dangerous and not needed at all.

File: src/util/db/fwdsqlquery.cpp [hunk 2]
```diff
@@ -94,33 +94,37 @@ DbFieldIndex FwdSqlQuery::fieldIndex(const QString& fieldName) const {
...
+constexpr int kBooleanTrue = 1;
+
+// NOTE(uklotzde): This conversion has been wrapped into a separate
+// function, because the conversion is completely independent of the
+// query, the current record and how values for individual fields are
+// retrieved. For separation of concerns and to improve readability.
+// Please do not try to unwrap the code! It is already declared "inline" ;)
+//
+// Recommended reading: "Refactoring" by Martin Fowler
+inline bool toBoolean(const QVariant& variant) {
```
Comment (Holzhaus) [inline src/util/db/fwdsqlquery.cpp:108]: boolFromQVariantInt?

Because this in only for ints. There's also a dedicated `variant.toBool(&ok)`.

**Review by Holzhaus — APPROVED** [review]
LGTM.

File: src/util/db/fwdsqlquery.cpp (hunk 2 above)
Comment (Holzhaus) [reply src/util/db/fwdsqlquery.cpp:108]: @uklotzde Should I merge as-is or should we clarify the naming first?

**Pull request merged**